src/admin/index.js

```diff
@@ -283,6 +284,63 @@ module.exports = ({
     })
   }
 
+  /**
+   * @param {string} topic
+   * @param {number=} timestamp
+   */
+
+  const fetchTopicOffsetsByTimestamp = async (topic, timestamp) => {
+    if (!topic || typeof topic !== 'string') {
+      throw new KafkaJSNonRetriableError(`Invalid topic ${topic}`)
+    }
+
+    const retrier = createRetry(retry)
+
+    return retrier(async (bail, retryCount, retryTime) => {
+      try {
+        await cluster.addTargetTopic(topic)
```

Comment:
We should not do this, since it means that the topic will be part of the targetTopics of the cluster forever, even if this was just a one-time operation. There is a `metadata` method on the cluster that allows you to get metadata for topics without adding them to the targetTopics:

https://github.com/tulios/kafkajs/blob/18fcb258f6194c6c75e51af69522853516174905/src/cluster/index.js#L140-L153

Reply:
I just used the same flow as `fetchTopicOffsets`, do I need to change both of them?

Reply:
Huh, you're right, so we've been doing that all along. Keep it as it is for now, and I can look into it separately later.